### PR TITLE
Advanced simulcast control

### PR DESCRIPTION
**Pull request by pe1ros**

### CODE REVIEW

File: ConferenceDemo/src/Core/Services/ConferenceService.ts [hunk 1]
```diff
@@ -62,7 +52,84 @@ export const ConferenceService = () => {
...
+    endpoint?.off();
+  };
+
+  const endConference = () => {
+    hangUp();
+  };
+
+  const hangUp = () => {
+    currentConference.current?.hangup();
+  };
```
Comment (YuliaGrigorieva) [inline ConferenceDemo/src/Core/Services/ConferenceService.ts]: Seems only one method is needed

File: ConferenceDemo/src/Core/Services/ConferenceService.ts [hunk 2]
```diff
@@ -62,7 +52,84 @@ export const ConferenceService = () => {
...
+    await currentConference.current.sendVideo(isSendVideo);
+  };
+
+  const streamManager = async (
+    count: number,
+    participant: IParticipant,
+    index: number,
+  ) => {
+    const endpoints = currentConference.current?.getEndpoints();
+    const findedEndoint = endpoints?.find(
```
Comment (YuliaGrigorieva) [inline ConferenceDemo/src/Core/Services/ConferenceService.ts:98]: finded -> found

File: ConferenceDemo/src/Core/Services/ConferenceService.ts [hunk 3]
```diff
@@ -62,7 +52,84 @@ export const ConferenceService = () => {
...
+    if (
+      !findedEndoint ||
+      !participant.streamId ||
+      currentConference.current.callId === participant.id
+    ) {
+      return;
+    }
+    if (count >= 6) {
+      if (index <= 5) {
+        if (!participant.isEnabledStream) {
```
Comment (YuliaGrigorieva) [inline ConferenceDemo/src/Core/Services/ConferenceService.ts:110]: isEnabledStream is better to change to hasEnabledStream

File: ConferenceDemo/src/Core/Services/HardwareService.ts [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+import {setListDevices, setSelectedDevice} from '../Store/conference/actions';
+
+export const HardwareService = () => {
+  const dispatch = useDispatch();
+
+  const AudioDeviceManager = useRef(
+    Voximplant.Hardware.AudioDeviceManager.getInstance(),
+  );
+  const CameraManager = Voximplant.Hardware.CameraManager.getInstance();
```
Comment (YuliaGrigorieva) [inline ConferenceDemo/src/Core/Services/HardwareService.ts:20]: Why AudioDeviceManager requires useRef, but CameraManager does not?

File: ConferenceDemo/src/Screens/Conference/index.tsx [hunk 5]
```diff
@@ -50,9 +52,12 @@ const ConferenceScreen = ({route}: IScreenProps<'Conference'>) => {
   const [modalVisible, setModalVisible] = useState(false);
 
   useEffect(() => {
-    startConference(conference, isSendVideo);
     getAudioDevices();
     getActiveDevice();
+    CameraManager.setCameraResolution(720, 480);
```
Comment (YuliaGrigorieva) [inline ConferenceDemo/src/Screens/Conference/index.tsx:57]: Please remove this line, SDK will use 640x480 as the default resolution

**Review by YuliaGrigorieva — CHANGES_REQUESTED** [review]

File: ConferenceDemo/src/Core/Services/ConferenceService.ts (hunk 1 above)
Comment (pe1ros) [reply ConferenceDemo/src/Core/Services/ConferenceService.ts]: done

File: ConferenceDemo/src/Core/Services/ConferenceService.ts (hunk 2 above)
Comment (pe1ros) [reply ConferenceDemo/src/Core/Services/ConferenceService.ts:98]: done

File: ConferenceDemo/src/Core/Services/ConferenceService.ts (hunk 3 above)
Comment (pe1ros) [reply ConferenceDemo/src/Core/Services/ConferenceService.ts:110]: done

File: ConferenceDemo/src/Screens/Conference/index.tsx (hunk 5 above)
Comment (pe1ros) [reply ConferenceDemo/src/Screens/Conference/index.tsx:57]: done

File: ConferenceDemo/src/Core/Services/HardwareService.ts (hunk 4 above)
Comment (pe1ros) [reply ConferenceDemo/src/Core/Services/HardwareService.ts:20]: done

**Review by YuliaGrigorieva — APPROVED** [review]

**Pull request merged**